src/main/java/uk/gov/hmcts/divorce/divorcecase/model/CaseDocuments.java

```diff
@@ -199,4 +200,12 @@ public Optional<ListValue<DivorceDocument>> getDocumentGeneratedWithType(final D
             .filter(document -> documentType.equals(document.getValue().getDocumentType())).findFirst()
             : Optional.empty();
     }
+
+    @JsonIgnore
+    public void removeDocumentGeneratedWithType(final DocumentType documentType) {
+        if (!isEmpty(this.getDocumentsGenerated())) {
+            this.getDocumentsGenerated()
+                .removeIf(document -> CONDITIONAL_ORDER_GRANTED.equals(document.getValue().getDocumentType()));
```

Comment:
```suggestion
                .removeIf(document -> documentType.equals(document.getValue().getDocumentType()));
```

Reply:
fixed in commit https://github.com/hmcts/nfdiv-case-api/pull/2112/commits/c5104f21b2769420a864bf5531c0b6f78c0a08a0